subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -991,9 +970,16 @@ public ArtifactView artifactView() {
 
 
             @Override
-            public ArtifactView attributes(Map<?, ?> attributeMap) {
+            public AttributeContainer getAttributes() {
+                return getViewAttributes();
+            }
+
+            @Override
+            public ArtifactView attributes(Action<? super AttributeContainer> action) {
                 assertUnset("attributes", viewAttributes);
-                this.viewAttributes = attributesFactory.fromPolymorphicMap(attributeMap);
+                DefaultMutableAttributeContainer mutableAttributes = new DefaultMutableAttributeContainer(attributesFactory);
```

Comment:
This is inconsistent with Configurations. I should be able to call this as many times as I wish and it should be locked once I call `getFiles` or `getArtifacts`